addons/udes_stock/tests/test_stock_picking.py

```diff
@@ -0,0 +1,289 @@
+from . import common
+from odoo.exceptions import UserError, ValidationError
+
+
+class TestStockPicking(common.BaseUDES):
+    @classmethod
+    def setUpClass(cls):
+        super(TestStockPicking, cls).setUpClass()
+        cls.Picking = cls.env['stock.picking']
+        products_info = [{'product': cls.apple, 'qty': 10}]
+        cls.test_picking_in = cls.create_picking(
+            cls.picking_type_goods_in, products_info=products_info, confirm=True,
+            location_dest_id=cls.test_received_location_01.id,
+        )
+        cls.test_picking_pick = cls.create_picking(
+            cls.picking_type_pick, products_info=products_info, confirm=True
+        )
+
+    def test01_get_empty_locations(self):
+        """ Empty locations - for goods in"""
+        self.assertEqual(
+            self.test_picking_in.get_empty_locations(),
+            self.test_received_location_01,
+        )
+        # Add stock to a location - to check empty locations obtained
+        self.create_quant(
+            self.apple.id, self.test_received_location_01.id,
+            5,
+        )
+        self.assertFalse(self.test_picking_in.get_empty_locations())
+
+    def test02_get_child_locations_simple_success(self):
+        """ Get child location"""
+        locations = self.test_goodsout_locations | self.out_location
+        self.assertEqual(self.test_picking_pick._get_child_dest_locations(), locations)
+
+    def test03_get_child_locations_simple_success_with_extra_domain(self):
+        """ Get child locations - with extra domain"""
+        aux_domain = [('name', '=', self.test_goodsout_location_01.name)]
+        self.assertEqual(
+            self.test_picking_pick._get_child_dest_locations(aux_domain=aux_domain),
+            self.test_goodsout_location_01,
+        )
+
+    def test04_get_child_locations_failure_with_extra_domain(self):
+        """ Return no child locations - with extra domain"""
+        aux_domain = [('name', '=', 'Not a location')]
+        self.assertFalse(self.test_picking_pick._get_child_dest_locations(aux_domain=aux_domain))
+
+    def test05_create_picking_no_moves(self):
+        """ Create a picking from picking type but have no moves created as no proudts_info given """
+        pick = self.Picking.create_picking(picking_type=self.picking_type_pick)
+        # Default names for pick location
```

Comment:
Revise comment, not checking names, checking default locations are the ones expected.